Mathlib/Order/BourbakiWitt.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2025 Finn Mortimore. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Finn Mortimore
+-/
+import Mathlib.Order.Preorder.Chain
+import Mathlib.Data.Set.Lattice
+import Mathlib.Dynamics.FixedPoints.Basic
+
+/-!
+# Bourbaki-Witt Theorem
+
+This file proves the Bourbaki-Witt Theorem.
+
+## Main definitions
+
+- class `ChainCompletePartialOrder` : A non-empty partial order is a chain complete partial order
+  such that every non-empty chain has a supremum
+- structure `Inflationary` : Let $X$ be a partial order. A function $f : X → X$ is inflationary if,
+  for all $x ∈ X$, $x ≤ f x$
+
+## Main statements
+
+- `bourbaki_witt` : The Bourbaki-Witt Theorem : If $X$ is a chain complete
+partial order and $f : X → X$ is inflationary, then $f$ has a fixed point
+
+## References
+
+The proof used can be found in [serge_lang_algebra]
+-/
+
+/-- The type of non-empty chains of an order -/
+@[ext]
+structure NonemptyChain (α : Type) [LE α] where
```

Comment:
Everywhere in this file, please generalize `Type` (which lives in one specific universe) to `Type*` (which allows any universe).
```suggestion
structure NonemptyChain (α : Type*) [LE α] where
```